src/print/fuchsia.rs

```diff
@@ -176,19 +171,14 @@ fn take_bytes_align4<'a>(num: usize, bytes: &mut &'a [u8]) -> Option<&'a [u8]> {
     Some(out)
 }
 
-// This function has no real invariants the caller must uphold other than
-// perhaps that 'bytes' should be aligned for performance (and on some
-// architectures correctness). The values in the Elf_Nhdr fields might
-// be nonsense but this function ensures no such thing.
+// This function has no invariants the caller must uphold, but it will return `None`
+// if 'bytes' is not correctly aligned.  The values in the Elf_Nhdr fields might be
```

Comment:
As long as we're here we might as well be exhaustive and make these doc comments.
```suggestion
/// This function has no invariants the caller must uphold, but
/// it will return `None`, without mutating, if `bytes` has insufficient size or alignment.
/// If this returns `Some(nhdr)`, then `bytes` was and remains 4-byte-aligned.
/// The values in the Elf_Nhdr fields might be nonsense.
```